src/corehost/cli/fxr/hostfxr.cpp

```diff
@@ -397,55 +395,451 @@ SHARED_API hostfxr_error_writer_fn hostfxr_set_error_writer(hostfxr_error_writer
+    if (context == nullptr)
+        return StatusCode::InvalidArgFailure;
+
+    if (context->type != host_context_type::initialized)
+    {
+        trace::error(_X("Setting properties is not allowed once runtime has been loaded."));
+        return StatusCode::InvalidArgFailure;
+    }
+
+    const corehost_context_contract &contract = context->hostpolicy_context_contract;
+    return contract.set_property_value(name, value);
 }
+
+//
+// Gets all the runtime properties for an initialized host context
+//
+// Parameters:
+//     host_context_handle
+//       Handle to the initialized host context
+//     count
+//       [in] Size of the keys and values buffers
+//       [out] Number of properties returned (size of keys/values buffers used). If the input value is too
+//             small or keys/values is nullptr, this is populated with the number of available properties
+//     keys
+//       Array of pointers to buffers with runtime property keys
+//     values
+//       Array of pointers to buffers with runtime property values
+//
+// Return value:
+//     The error code result.
+//
+// The buffers pointed to by keys and values are owned by the host context. The lifetime of the buffers is only
+// guaranteed until any of the below occur:
+//   - a 'run' method is called for the host context
+//   - properties are changed via hostfxr_set_runtime_property_value
+//   - the host context is closed via 'hostfxr_close'
+//
+// If host_context_handle is nullptr and an active host context exists, this function will get the
+// properties for the active host context.
+//
+SHARED_API int32_t __cdecl hostfxr_get_runtime_properties(
+    const hostfxr_handle host_context_handle,
+    /*inout*/ size_t * count,
+    /*out*/ const pal::char_t **keys,
+    /*out*/ const pal::char_t **values)
+{
+    trace_hostfxr_entry_point(_X("hostfxr_get_runtime_properties"));
+
+    if (count == nullptr)
+        return StatusCode::InvalidArgFailure;
+
+    const host_context_t *context;
+    if (host_context_handle == nullptr)
+    {
+        const host_context_t *context_maybe = fx_muxer_t::get_active_host_context();
+        if (context_maybe == nullptr || context_maybe->type != host_context_type::active)
+        {
+            trace::error(_X("Hosting components context has not been initialized. Cannot get runtime properties."));
+            return StatusCode::HostInvalidState;
+        }
+
+        context = context_maybe;
+    }
+    else
+    {
+        context = host_context_t::from_handle(host_context_handle);
+        if (context == nullptr)
+            return StatusCode::InvalidArgFailure;
+    }
+
+    if (context->type == host_context_type::secondary)
+    {
+        const std::unordered_map<pal::string_t, pal::string_t> &properties = context->config_properties;
+        size_t actualCount = properties.size();
+        if (*count < actualCount || keys == nullptr || values == nullptr)
+        {
+            *count = actualCount;
```

Comment:
The `*count` has to be set always, on success it must contain the number of properties populated (otherwise the caller would not know)